lib/elixir/src/elixir_with.erl

```diff
@@ -55,11 +55,14 @@ expand({'<-', Meta, [Left, Right]}, E) ->
 expand(X, E) ->
   elixir_exp:expand(X, E).
 
-expand_else(KV, E) when is_list(KV) ->
+expand_else(_Meta, KV, E) when is_list(KV) ->
   {[{do, EClauses}], EC} = elixir_exp_clauses:'case'([], [{do, KV}], E),
   {[{else, EClauses}], EC};
-expand_else(nil, E) ->
-  {[], E}.
+expand_else(_Meta, nil, E) ->
+  {[], E};
+expand_else(Meta, _KV, E) ->
+  elixir_errors:compile_error(Meta, ?m(E, file),
+  "expected -> clauses in else block", []).
```

Comment:
Good point @michalmuskala, the message for `case`s is 

> expected -> clauses for do in case

so maybe we can drop the "one or more" but we should say "in with's else block" or something along those lines IMO.